src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -103,24 +97,24 @@
    * @param compPOL   PO line to retrieve/create Item Records for. At this step PO Line must contain instance Id
    * @return future with list of pieces with item and location id's
    */
-  public CompletableFuture<List<Piece>> handleItemRecords(CompositePoLine compPOL) {
+  public CompletableFuture<List<Piece>> handleHoldingRecords(CompositePoLine compPOL) {
     List<CompletableFuture<List<Piece>>> itemsPerHolding = new ArrayList<>();
-
     // Group all locations by location id because the holding should be unique for different locations
-    groupLocationsById(compPOL)
-      .forEach((locationId, locations) -> {
-        int expectedQuantity = calculateInventoryItemsQuantity(compPOL, locations);
-        // For some cases items might not be created e.g. Electronic resource with create inventory set to false
-        if (expectedQuantity > 0) {
-          itemsPerHolding.add(
-            // Search for or create a new holding and then create items for this holding
-            getOrCreateHoldingsRecord(compPOL, locationId)
-              .thenCompose(holdingId -> handleItemRecords(compPOL, holdingId, expectedQuantity))
-              .thenApply(itemIds -> constructPieces(itemIds, compPOL.getId(), locationId))
-          );
-        }
-      });
-
+    if (HelperUtils.isHoldingsUpdateRequired(compPOL)) {
+      groupLocationsById(compPOL)
+        .forEach((locationId, locations) -> itemsPerHolding.add(
+          // Search for or create a new holding and then create items for this holding
+          getOrCreateHoldingsRecord(compPOL, locationId)
+            .thenCompose(holdingId -> {
+              int expectedQuantity = calculateInventoryItemsQuantity(compPOL, locations);
+              if (isItemsUpdateRequired(compPOL) && expectedQuantity > 0) {
+                  // For some cases items might not be created e.g. resources with create inventory set to "None"
+                  return handleItemRecords(compPOL, holdingId, expectedQuantity)
+                    .thenApply(itemIds -> constructPieces(itemIds, compPOL.getId(), locationId));
+                } else return completedFuture((new ArrayList<>()));
```

Comment:
Please use curly bracket notation, even for one-liners like this.  It improves readability

Reply:
also you indentation might be a little off here, can't tell in github

Reply:
added brackets. formatted code